发websocket连接请求后panic

proxy.exe --addr=127.0.0.1:8080 --addr-rpc=127.0.0.1:9091 --addr-store=etcd://127.0.0.1:2379 --namespace=test --websocket
proxy已经设置WebSocket参数

以下为Panic内容
panic: interface conversion: interface is nil, not http.ResponseWriter

goroutine 162 [running]:
github.com/fagongzi/gateway/pkg/proxy.(*Proxy).onWebsocket(0xc0001b8000, 0xc000684300, 0xc00002b310, 0xe, 0x0, 0x0, 0x0)
	D:/workspace/github/manba/pkg/proxy/proxy_websocket.go:114 +0x66a
github.com/fagongzi/gateway/pkg/proxy.(*Proxy).doProxy(0xc0001b8000, 0xc000410100, 0x0)
	D:/workspace/github/manba/pkg/proxy/proxy.go:477 +0x229e
github.com/fagongzi/gateway/pkg/proxy.(*Proxy).ServeFastHTTP(0xc0001b8000, 0xc0000f2d80)
	D:/workspace/github/manba/pkg/proxy/proxy.go:309 +0xfc5
github.com/valyala/fasthttp.(*Server).serveConn(0xc0000ee240, 0x170abe0, 0xc000414620, 0x0, 0x0)
	C:/Users/Administrator/go/pkg/mod/github.com/valyala/fasthttp@v1.2.0/server.go:1932 +0xd55
github.com/valyala/fasthttp.(*workerPool).workerFunc(0xc000436000, 0xc000004040)
	C:/Users/Administrator/go/pkg/mod/github.com/valyala/fasthttp@v1.2.0/workerpool.go:212 +0x110
github.com/valyala/fasthttp.(*workerPool).getCh.func1(0xc000436000, 0xc000004040, 0x149dec0, 0xc000004040)
	C:/Users/Administrator/go/pkg/mod/github.com/valyala/fasthttp@v1.2.0/workerpool.go:184 +0x3c
created by github.com/valyala/fasthttp.(*workerPool).getCh
	C:/Users/Administrator/go/pkg/mod/github.com/valyala/fasthttp@v1.2.0/workerpool.go:183 +0x1fc

--- a/pkg/proxy/proxy_start.go
+++ b/pkg/proxy/proxy_start.go
@@ -164,8 +164,8 @@ func (p *Proxy) startHTTPCMUX() {
 	}
 
 	m := cmux.New(l)
-	go p.startHTTPWithListener(m.Match(cmux.Any()))
 	go p.startHTTPWebSocketWithListener(m.Match(cmux.HTTP1HeaderField("Upgrade", "websocket")))
+	go p.startHTTPWithListener(m.Match(cmux.Any()))
 	err = m.Serve()
 	if err != nil {
 		log.Fatalf("start http failed failed with %+v",

